soc/arm/nordic_nrf/common/soc_nrf_common.S

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for Nordic Semiconductor nRFxx family processors
+ */
+
+#include <toolchain.h>
+
+_ASM_FILE_PROLOGUE
+
+GTEXT(SystemInit)
+GTEXT(z_platform_init)
+
+SECTION_FUNC(TEXT, z_platform_init)
+
+#if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
+	ldr r0, =SystemInit
+	bx r0
+#elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+	b SystemInit
+#endif
```

Comment:
An `#else` here would help prevent `z_platform_init` from ending up without any implementation in an unsupported configuration.

Reply:
I know... I also thought about this, right. But currently nRF families only have cortex-m, and cortex-m only has these 3 ARM architecture variants, so I was a bit reluctant to add and #error "unsupported arch".  :) 
But I can add it now that you also suggested this. no big deal